### PR TITLE
[feat] initial dynamodb support

### PR DESCRIPTION
**Pull request by adiom-mark**
Currently a rough prototype for dynamodb, support as a source, and as a destination its initial sync only to bootstrap

To try it out you need to set up localstack.

You need to create a dynamodb table using the cli. Then you can try using testconn as a source and feed data into dynamodb.

Then you can use dynamodb as a source to feed data into mongo.

### CODE REVIEW

File: connectors/dynamodb/stream/helper.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	})
+	return err
+}
+
+func (s *dynamodbStreamHelper) Wait() error {
+	<-s.done
+	return s.err
+}
+
+func (s *dynamodbStreamHelper) GetShards(ctx context.Context) ([]types.Shard, error) {
```
Comment (alex-thc) [inline connectors/dynamodb/stream/helper.go:98]: Nitpick: There's a function with the exact same name in the same file but lower-case first letter, yet the way they do things is very different. Can the function name be more descriptive?

File: connectors/dynamodb/stream/helper.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+	}
+
+	streamARN := aws.String(s.streamARN)
+	state := NewStreamState(s.streamARN)
+
+	eg, egCtx := errgroup.WithContext(ctx)
+
+	for _, shard := range openShards {
+		eg.Go(func() error {
+			shardIteratorRes, err := s.streamClient.GetShardIterator(egCtx, &dynamodbstreams.GetShardIteratorInput{
```
Comment (alex-thc) [inline connectors/dynamodb/stream/helper.go:291]: when does this start reading? At the beginning of time?

**Review by alex-thc — APPROVED** [review]

File: connectors/dynamodb/stream/helper.go (hunk 1 above)
Comment (adiom-mark) [reply connectors/dynamodb/stream/helper.go:98]: this is the public one, under the hood it ends up using the other one

File: connectors/dynamodb/stream/helper.go (hunk 2 above)
Comment (adiom-mark) [reply connectors/dynamodb/stream/helper.go:291]: it starts reading from the earliest entries in the shards that are still open (no ending sequence number)

**Comment by alex-thc** [comment]
But this could be 24hr ago, right? So worst case scenario it’ll take time
to finish?

Alex

**Comment by adiom-mark** [comment]
aws documentation says they rotate shards every 4 hours or so, so a few hours probably

**Pull request merged**